### ISSUE TITLE
filterConfig not filtering methods with attribute

### ISSUE THREAD

**Issue by jalley3** [dotnet/docfx#7064]
**Operating System**: (`Windows` or `Linux` or `MacOS`)

windows

**DocFX Version Used**: 2.56.6

**Steps to Reproduce**:

1. Filter config file:
```
apiRules:
- exclude:
    hasAttribute:
        uid: System.ComponentModel.EditorBrowsableAttribute
        ctorArguments:
        - System.ComponentModel.EditorBrowsableState.Advanced
```
2.  Attribute over code:
`[System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Advanced)]`

3. Metadata section of docfx.json

```
"metadata": [
    {
      "src": [
        {
          "files": [
            "src/**/**.dll"
          ],
          "exclude": [
            "**/obj/**",
            "_site/**"
          ]
        }
      ],
      "dest": "api-reference",
      "filter": "filterConfig.yml"
    }
],
```

**Expected Behavior**:
The methods that are decorated with that attribute be filtered out from a class's methods collection

**Actual Behavior**:
The internal methods decorated by the attribute described above are still present on the class' methods collections

**Comment by jalley3** [comment]
Cloning the repo and debugging manually it does not appear as though the ctor argument is even being captured. In the AttributeFilterInfo class, when you inspect the symbol's attribute collection, it shows the System.ComponentModel.EditorBrowsable attribute but if you look at the constructor collections both are empty:

![image](https://user-images.githubusercontent.com/17911817/108094946-53768100-704d-11eb-9173-5eea35feaafe.png)

**Comment by v-pegao** [comment]
The issue should be fixed in [version 2.56.7 of docfx](https://www.nuget.org/packages/docfx/2.56.7), would you please have another try on version 2.56.7? Related change is [here](https://github.com/dotnet/docfx/pull/7007)

**Comment by bitbonk** [comment]
@v-pegao This issue still exists or is back: #7815
